test/spec/checks_spec.js

```diff
@@ -0,0 +1,153 @@
+describe('Checks Spec', function(){
+    function runCheckSpec (context) {
+        describe(context.checkId, function(){
+            var rules = customRules,
+                checkId = context.checkId,
+                fixture = context.fixture,
+                selector = context.selector,
+                expectedResult = context.result,
+                checkContext = {
+                    _relatedNodes: [],
+                    _data: null,
+                    data: function (d) {
+                        this._data = d;
+                    },
+                    relatedNodes: function (rn) {
+                        this._relatedNodes = rn;
+                    }
+                },
+                results;
+
+            var findCheck = function(){
+                var filtered = rules.checks.filter(function(value){
+                    return value.id === checkId;
+                });
+                expect(filtered.length).toEqual(1);
+                return filtered[0];
+            };
+
+            var evaluateCheck = function(){
+                check = findCheck(checkId);
```

Comment:
Same `var` recommendation as above for `check` and `nodes`--functional scoping here would make sense.